### ISSUE TITLE
As a service author I would like all types passed through broker extension points to have a sensible `toString` method

### ISSUE THREAD

**Issue by spikymonkey** [spring-cloud-incubator/spring-cloud-app-broker#90]
...so I can understand Matcher failures in my tests :)

Here's an example using a custom matcher we wrote. Because `BackingApplication` doesn't have an overridden `toString`, it's impossible to work out what's wrong without using the debugger.

```
io.pivotal.spring.cloud.configserver.servicebroker.ServiceBrokerComponentTest > createMirrorsMultipleGitReposInComposite() FAILED
    Argument(s) are different! Wanted:
    backingAppDeploymentService bean.deploy(
        BackingApplications containing a single app with environment equal to: <{spring.cloud.config.server.composite[0].cloneOnStart=true, spring.cloud.config.server.composite[0].uri=http://mirror.example.com/fortune-teller, spring.cloud.config.server.composite[1].label=myLabel, spring.cloud.config.server.composite[1].type=git, spring.cloud.config.server.composite[1].uri=http://mirror.example.com/fortune-teller2, spring.cloud.config.server.composite[0].label=myLabel, spring.cloud.config.server.composite[0].type=git, spring.cloud.config.server.composite[1].cloneOnStart=true}>
    );
    -> at io.pivotal.spring.cloud.configserver.servicebroker.ServiceBrokerComponentTest.createMirrorsMultipleGitReposInComposite(ServiceBrokerComponentTest.java:111)
    Actual invocation has different arguments:
    backingAppDeploymentService bean.deploy(
        [org.springframework.cloud.appbroker.deployer.BackingApplication@382b4f80]
    );
```

**Comment by scottfrederick** [comment]
Fixed in 986d72cba157429c181c12e12a844185e92fd870

**Comment by spikymonkey** [comment]
Awesome thanks @scottfrederick !
